drivers/adc/ad9083/ad9083.c

```diff
@@ -0,0 +1,394 @@
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <errno.h>
+#include <stdlib.h>
+#include "ad9083.h"
+#include "error.h"
+#include <inttypes.h>
+#include "delay.h"
+#include "clk.h"
+#include "uc_settings.h"
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+extern int32_t adi_ad9083_jtx_startup(adi_ad9083_device_t *device,
+				      adi_cms_jesd_param_t *jtx_param);
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define CHIPID_AD9083		0x00EA
+#define CHIPID_MASK		0xFFFF
+#define SPI_IN_OUT_BUFF_SZ	0x3
+#define MAX_REG_ADDR		0x1000
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Read register.
+ * @param device - ad9083 device.
+ * @param reg - Address of the register.
+ * @param readval - Value of the register.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad9083_reg_get(struct ad9083_phy *device, uint32_t reg,
+		       uint8_t *readval)
+{
+	int32_t ret;
+	uint8_t data[SPI_IN_OUT_BUFF_SZ] = {0};
+
+	if (reg < MAX_REG_ADDR) {
+		data[0] = (reg >> 8) | 0x80;
```

Comment:
I guess `0x80` corresponds to the read command. 

Therefore, a macro would be useful.